File: foundation-domain/ai-platform/Backend/Java/ai-monitoring-service/src/main/java/com/gogidix/aiservices/aimonitoringservice/shared/exception/ConflictException.java
package com.gogidix.aiservices.aimonitoringservice.shared.exception;

/**
 * Exception thrown when a conflict occurs.
 */
public class ConflictException extends BaseDomainException {

    public ConflictException(String message) {
        super(message);
    }

    @Override
    protected String deriveErrorCode() {
        return "CONFLICT";
    }
}

File: management-domain/Sales-department/Backend/Nodes/sales-automation-service/commission-calculator-service/src/shared/exceptions/conflict.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Conflict exception
 * Thrown when a conflict with existing data occurs
 */
export class ConflictException extends BaseException {
  readonly errorCode = 'CONFLICT';

  constructor(message: string, context?: Record<string, unknown>) {
    super(message, HttpStatus.CONFLICT, context);
  }
}
